chain/client/src/stateless_validation/state_witness_producer.rs

```diff
@@ -78,12 +79,30 @@ impl Client {
             chunk.chunk_hash(),
             chunk_validators,
         );
+
+        // Record the witness in order to match the incoming acks for measuring round-trip times.
+        // See process_chunk_state_witness_ack for the handling of the ack messages.
+        self.state_witness_tracker.record_witness_sent(
+            &witness,
+            witness_size,
+            chunk_validators.len(),
+        );
+
         self.network_adapter.send(PeerManagerMessageRequest::NetworkRequests(
             NetworkRequests::ChunkStateWitness(chunk_validators, witness),
         ));
         Ok(())
     }
 
+    /// Handles the state witness ack message from the chunk validator.
```

Comment:
```suggestion
    /// Handles the state witness ack message from the chunk validator.
    ///
```

and more generally there's some excellent documentation on writing documentation strings [here](https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html#how-to-write-documentation).